test/lorenz_delta.saved

```diff
@@ -0,0 +1,239 @@
+Starting braid_TestStepDiff
+   braid_TestStepDiff:   Starting Test 1
+   braid_TestStepDiff:   A = clone(B) 
+   braid_TestStepDiff:   v = clone(u) 
+   braid_TestStepDiff:   v = step(v)
+   braid_TestStepDiff:   B = step_dv(v)*B
+   braid_TestStepDiff:   u = step(u + eps*A_0) 
+   braid_TestStepDiff:   v = B_0 - (u - v)/eps
+   braid_TestStepDiff:  (v = step_dv(u)*A_0 - (step(u + eps*A_0) - step(u))/eps) 
+   braid_TestStepDiff:   result = inner_prod(v, v) 
+   braid_TestStepDiff:   actual output:    result approx. 6.57e-19 
+   braid_TestStepDiff:   expected output:  positive, near zero 
+Finished braid_TestStepDiff
+Starting braid_TestBufBasis
+   braid_TestBufBasis:   size, size_basis = bufsize()
+   braid_TestBufBasis:   buffer = malloc(size + rank * size_basis)
+   braid_TestBufBasis:   buffer[:size] = bufpack(u)
+   braid_TestBufBasis:   buffer[size:] = bufpack(A)
+   braid_TestBufBasis:   v = bufunpack(buffer[:size]))
+   braid_TestBufBasis:   B = bufunpack(buffer[size:]))
+   braid_TestBufBasis:   v = u - v
+   braid_TestBufBasis:   B = A - B
+   braid_TestBufBasis:   inner_prod(v, v)
+   braid_TestBufBasis:   Test 1 Passed
+   braid_TestBufBasis:   actual output:    inner_prod(v, v) = 0.00e+00  
+   braid_TestBufBasis:   expected output:  inner_prod(v, v) = 0.0 
+   braid_TestBufBasis:   Frobenius_norm(B)
+   braid_TestBufBasis:   Test 2 Passed
+   braid_TestBufBasis:   actual output:    Frobenius_norm(B) = 0.00e+00  
+   braid_TestBufBasis:   expected output:  Frobenius_norm(B) = 0.0 
+Finished braid_TestBufBasis
+   braid_TestDelta:   free(u) 
+   braid_TestDelta:   free(v) 
+   braid_TestDelta:   free(A) 
+   braid_TestDelta:   free(B) 
+Finished braid_TestDelta
+
+# Begin Test 1
+
+  residual norm         = 6.398782e-07
+
+# Begin Test 2
+
+  residual norm         = 4.386838e-07
+  Average Lyapunov Exponents: [0.008363]
+
+# Begin Test 3
+
+  residual norm         = 5.788984e-08
+  Average Lyapunov Exponents: [0.008363, -0.417301]
+
+# Begin Test 4
+
+  residual norm         = 5.533993e-14
+  Average Lyapunov Exponents: [0.008363, -0.417301, -13.208420]
+
+# Begin Test 5
+
+  residual norm         = 1.470200e-07
+
+# Begin Test 6
+
+  residual norm         = 7.259506e-07
+  Average Lyapunov Exponents: [0.008363]
+
+# Begin Test 7
+
+  residual norm         = 4.441509e-07
+  Average Lyapunov Exponents: [0.008363, -0.417301]
+
+# Begin Test 8
+
+  residual norm         = 5.564901e-14
+  Average Lyapunov Exponents: [0.008363, -0.417301, -13.208420]
+
+# Begin Test 9
+
+  residual norm         = 7.940954e-07
+  Average Lyapunov Exponents: [0.000000]
+
+# Begin Test 10
+
+  residual norm         = 3.460783e-07
+  Average Lyapunov Exponents: [0.000000, 0.000000]
+
+# Begin Test 11
+
+  residual norm         = 3.147082e-07
+  Average Lyapunov Exponents: [0.000000, 0.000000, 0.000000]
+
+# Begin Test 12
+
+  residual norm         = 8.077582e-07
+  Average Lyapunov Exponents: [0.008363]
+
+# Begin Test 13
```

Comment:
I got about 5 tests with sometimes tiny differences like
```
--- lorenz_delta.saved.13       2023-02-10 16:29:19.933682822 -0700
+++ std.out.13  2023-02-10 16:29:22.172290355 -0700
@@ -1,5 +1,2 @@
-# Begin Test 13
-
-  residual norm         = 7.293948e-08
+  residual norm         = 7.293949e-08
   Average Lyapunov Exponents: [0.008363, -0.417301]
-
```

and sometimes larger differences like

```
jacobschroder$ more std.err.17
--- lorenz_delta.saved.17       2023-02-10 16:29:19.933880695 -0700
+++ std.out.17  2023-02-10 16:29:22.773064267 -0700
@@ -1,5 +1,2 @@
-# Begin Test 17
-
-  residual norm         = 5.637681e-14
+  residual norm         = 5.294091e-14
   Average Lyapunov Exponents: [0.000000, 0.000000, 0.000000]
-
jacobschroder$ more std.err.18
--- lorenz_delta.saved.18       2023-02-10 16:29:19.933933611 -0700
+++ std.out.18  2023-02-10 16:29:22.923200818 -0700
@@ -1,5 +1,2 @@
-# Begin Test 18
-
-  residual norm         = 5.181244e-14
+  residual norm         = 4.707584e-14
   Average Lyapunov Exponents: [-8.416132, -1.601542, -3.599683]
```

Overall, the regression tests look great!  

But maybe we should try limiting the number of XBraid iters to 2 or 3 in all the tests?  I think this will leave the resid norm larger (and less prone to loss of precision).  

Fewer time steps could also help, but if we drive the residual to close to 0, this may not help. 

Let's iterate a few times, trying first limiting the number of iterations, until my machine and yours give the same reg test numbers!